waddrmgr/db.go

```diff
@@ -484,36 +790,86 @@ func putAccountInfo(tx walletdb.Tx, account uint32, encryptedPubKey,
 		acctType: actBIP0044,
 		rawData:  rawData,
 	}
-	return putAccountRow(tx, account, &acctRow)
+	if err := putAccountRow(tx, account, &acctRow); err != nil {
+		return err
+	}
+	// Update account id index
+	if err := putAccountIdIndex(tx, account, name); err != nil {
+		return err
+	}
+	// Update account name index
+	if err := putAccountNameIndex(tx, account, name); err != nil {
+		return err
+	}
+	return nil
 }
 
-// fetchNumAccounts loads the number of accounts that have been created from
-// the database.
-func fetchNumAccounts(tx walletdb.Tx) (uint32, error) {
-	bucket := tx.RootBucket().Bucket(acctBucketName)
+// putNumAllAddrs stores the provided metadata - number of new addrs - to the database.
+func putNumAllAddrs(tx walletdb.Tx, count uint32) error {
```

Comment:
The behavior of this put should not be different from all the rest.  In particular, the existing put functions take the specific value to put and so should this one.  The problem with the current approach of taking a count and incrementing it is this means, it's impossible to delete addresses, which needs to be supported.  It will be common to allow watch-only addresses to be deleted, for example.